src/pages/workspace/WorkspaceSettingsCurrencyPage.tsx

```diff
@@ -1,64 +1,61 @@
-import PropTypes from 'prop-types';
 import React, {useCallback, useState} from 'react';
 import {withOnyx} from 'react-native-onyx';
-import _ from 'underscore';
+import type {OnyxEntry} from 'react-native-onyx';
 import FullPageNotFoundView from '@components/BlockingViews/FullPageNotFoundView';
 import HeaderWithBackButton from '@components/HeaderWithBackButton';
 import ScreenWrapper from '@components/ScreenWrapper';
 import SelectionList from '@components/SelectionList';
 import useLocalize from '@hooks/useLocalize';
-import compose from '@libs/compose';
 import Navigation from '@libs/Navigation/Navigation';
 import * as PolicyUtils from '@libs/PolicyUtils';
 import * as Policy from '@userActions/Policy';
 import ONYXKEYS from '@src/ONYXKEYS';
 import ROUTES from '@src/ROUTES';
-import {policyDefaultProps, policyPropTypes} from './withPolicy';
+import type {Currency} from '@src/types/onyx';
+import {isEmptyObject} from '@src/types/utils/EmptyObject';
+import type {WithPolicyAndFullscreenLoadingProps} from './withPolicyAndFullscreenLoading';
 import withPolicyAndFullscreenLoading from './withPolicyAndFullscreenLoading';
 
-const propTypes = {
+type WorkspaceSettingsCurrentPageWithOnyxProps = {
     /** Constant, list of available currencies */
-    currencyList: PropTypes.objectOf(
-        PropTypes.shape({
-            /** Symbol of the currency */
-            symbol: PropTypes.string.isRequired,
-        }),
-    ),
-    isLoadingReportData: PropTypes.bool,
-    ...policyPropTypes,
+    currencyList: OnyxEntry<Record<string, Currency>>;
```

Comment:
```suggestion
    currencyList: OnyxEntry<CurrencyList>;
```

Could you create a `CurrencyList` type in `src/types/onyx/Currency.ts` and use it here and [here](https://github.com/Expensify/App/blob/main/src/ONYXKEYS.ts#L379)?

```ts
type CurrencyList = Record<string, OnyxTypes.Currency | null>;
```